agent/eventhandler/task_handler.go

```diff
@@ -217,6 +217,17 @@ func (handler *TaskHandler) SubmitTaskEvents(taskEvents *eventList, client api.E
 					seelog.Debugf("TaskHandler, Submitted task state change: %s", event.String())
 					backoff.Reset()
 					taskEvents.events.Remove(eventToSubmit)
+
+					// ExecutionStoppedAt can also be sent here
+					if event.taskChange.ExecutionStoppedAt != nil {
```

Comment:
This logic is getting complicated. If we want to track a couple more timestamps, it'll get even more complicated. Instead of this, let's just have a boolean for each kind of timestamp that we track and have a `Sent` status field to track it.